### ISSUE TITLE
Between tag missing pagination

### ISSUE THREAD

**Issue by enxoco** [transformstudios/statamic-events#18]
The between tag seems to be working as expected now except for the fact that it seems to be missing pagination.

**Comment by enxoco** [comment]
@edalzell Currently doing this to get my pagination work.  Probably a little hacky but working for me at the moment.  I just had to modified my template slightly.

```php
public function between()
    {
        $from = $this->params->pull('from');
        $to = $this->params->pull('to');
        
        $this->loadEvents($this->params->bool('collapse_multi_days', false));

        if (! $from) {
            Log::error('Missing `from` parameter in `events:between` tag');

            return;
        }

        if (! $to) {
            Log::error('Missing `to` parameter in `events:between` tag');

            return;
        }

        $from = Carbon::parse($from)->startOfDay();
        $to = Carbon::parse($to)->endOfDay();
        $page = (int) request('page', 1);
        $this->limit = $this->params->int('limit', 1);
        $this->offset = $this->params->int('offset', 0);

        $this->offset = (($page - 1) * $this->limit) + $this->offset;

        $this->loadDates($from, $to);
        $paginator = new LengthAwarePaginator(
            array_slice($this->dates->toArray(), $this->offset, $this->limit),
            15,
            $this->limit,
            $page
        );

        $paginator->setPath(URL::makeAbsolute(URL::getCurrent()));
        $paginator->appends(request()->all());
        $links = [];
        for ($i = 1; $i <= $paginator->lastPage(); $i++)
        {
            array_push($links, $i);
        }
        $myDates = array_values(array_merge(
            $this->makeEmptyDates($from, $to),
            array_slice($this->dates->toArray(), $this->offset, $this->limit)));

        
        return array_values(array_merge(
            $this->makeEmptyDates($from, $to),
            array_slice($this->dates->toArray(), $this->offset, $this->limit),
            ['paginate' => [
                'total_items'    => count($myDates),
                'items_per_page' => 5,
                'total_pages'    => $paginator->lastPage(),
                'current_page'   => $paginator->currentPage(),
                'prev_page'      => $paginator->previousPageUrl(),
                'next_page'      => $paginator->nextPageUrl(),
                'links'          => $paginator->getUrlRange(1, $paginator->lastPage()),
            ]
            ]
        ));

    }
```

**Comment by enxoco** [comment]
@edalzell Any updates on this?

**Comment by edalzell** [comment]
@enxoco question for you. Both `in` and `between` return "grouped" (by day) results. How would you want to paginate those? Like if you say `limit=3` and there are 2 days of events, but the first day has 5 events, what would you want?

**Comment by enxoco** [comment]
@edalzell Good question.  I think for our current use case I would just want the events in order by date.  So if there are 2 days of events so for example:

Page 1
day 1 - event 1
day 1 - event 2
day 1 - event 3
---- Page 2
day 1 - event 4
day 1 - event 5
day 2 - event 1

**Comment by edalzell** [comment]
Right @enxoco but what would the tag look like that would produce that? What would `limit` be?

**Comment by enxoco** [comment]
@edalzell It looks like this
```      
{{
        events:between
        collection="events"
        limit="5"
        paginate="true"
        :taxonomy:event_categories="get:cat"
        :from="get:from"
        :to="get:to"
        :query="get:q"
        filter="calendar_search"
      }}
```

**Comment by edalzell** [comment]
Right, but like I said, what does that "5" mean? 5 days? 5 events? What if there are 2 events on 1 day and 6 on a second day? How would that paginate?

I "think" pagination only works when the events aren't grouped by day, like the `upcoming` tag.

**Comment by enxoco** [comment]
@edalzell This is how the old version of the plugin works for reference.  This is what we need the new version to do.

https://www.palmettodunes.com/calendar?from=2022-01-11&to=2022-01-18&q=&cat%5B%5D=pickleball&cat%5B%5D=golf&cat%5B%5D=tennis

**Comment by edalzell** [comment]
Right that's not grouped by date.

Thinking more, I don't think it should group by date at all, except in the case of the calendar tag.

**Comment by enxoco** [comment]
@edalzell I really just need the new version of the between tag to work like it does in the older version of the plugin as seen on the page referenced above.

**Comment by enxoco** [comment]
@edalzell I guess my other question then is, is there a better way to do what I am trying to do with the new plugin?  Just need to give the user the ability to search for events by date and category and display the results with pagination.

**Comment by edalzell** [comment]
Released in 3.3 @enxoco

**Comment by enxoco** [comment]
@edalzell I must be missing something here.  I just created a fresh install of Statamic, installed the latest version of the plugin and copied over my events.  This is what I get when using your example code from the docs.
```
{{ events:between collection="events" from="Jan 1 2022" to="March 31 2022" }}
  {{ date }} {{# date of event #}}
  {{ if no_results }}
    {{# whatever you need to when for an empty day #}}
  {{ else }}
    {{ dates }}
      ...other entry data...
      {{ date }}
    {{ /dates }}
  {{ /if }}
{{ /events:between }}
```
![image](https://user-images.githubusercontent.com/34722899/149207406-3d630a3a-e62a-48da-ad9f-4f571a73b46c.png)

As you can see, it appears that limit="2" only returns 2 results.  If I change it to 20 then I get 20 results all on one page.  How can I get all of the dates in that range and only show 5 per page?

![image](https://user-images.githubusercontent.com/34722899/149207845-3df2b8b2-475d-4745-918b-a145d1d4e122.png)
